src/ttcontrol/ttcontrol.py

```diff
@@ -23,3 +38,50 @@ def read_rom():
     print("commit=a1b2c3d4")
     # TODO: if we have a ROM, read it and print it out
     # TODO: if we don't have a ROM (tt03p5), read the ROM data "rom.txt" and print it out
+
+
+def _get_best_rp2040_freq(freq, max_rp2040_freq):
+    # Scan the allowed RP2040 frequency range for a frequency
+    # that will divide to the target frequency well
+    min_rp2040_freq = 48_000_000
+
+    if freq > max_rp2040_freq // 2:
+        raise ValueError("Requested frequency too high")
+    if freq <= min_rp2040_freq // (2**24-1):
+        raise ValueError("Requested frequency too low")
```

Comment:
That's about 2 Hz. In theory, we could go slower using software - not sure if it's worth implementing ATM.